artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationResponseMessageV2.java

```diff
@@ -62,10 +62,9 @@ public void decodeRest(final ActiveMQBuffer buffer) {
    }
 
    @Override
-   public String toString() {
-      StringBuffer buf = new StringBuffer(getParentString());
+   protected String getPacketString() {
+      StringBuffer buf = new StringBuffer(getPacketString());
```

Comment:
This just introduced the very type of recursion the change was looking to prevent. Will send a PR to fix.